juju/juju
# Introduce state.IAASModel

## Description of change

This is the first of a series of changes that moves IAAS specific functionality from `state.State` onto a new `state.IAASModel` type. This is in preparation for new model types. The idea is that `State` will only deal with generic functionality with model type specific functionality being available via new types accessible via `State`.

This change moves everything to do with storage onto a new `IAASModel` type. Future PRs will move more areas of functionality across.

## QA steps

All unit tests pass. Deploying a simple storage charm works as expected:
```
$ juju deploy cs:~axwalk/trusty/storagetest-1 --storage filesystem=tmpfs --storage block=loop
$ juju ssh 0 -- mount  | grep /srv
filesystem-0-0 on /srv/filesystem/1 type tmpfs (rw,size=1024m)
```

## Documentation changes

N.A. - internal only.

## Bug reference

N.A.
